src/main/java/org/cbioportal/web/columnar/StudyViewColumnStoreController.java

```diff
@@ -308,8 +304,18 @@ public ResponseEntity<DensityPlotData> fetchClinicalDataDensityPlot(
             .yAxisLogScale(yAxisLogScale)
             .build();
         
-        List<ClinicalData> sampleClinicalDataList = studyViewColumnarService.getSampleClinicalData(interceptedStudyViewFilter, xyAttributeId);
-        DensityPlotData result = clinicalDataDensityPlotService.getDensityPlotData(sampleClinicalDataList, densityPlotParameters, interceptedStudyViewFilter);
+        List<ClinicalData> combinedClinicalDataList = fetchClinicalDataForXyPlot(
+            studyViewColumnarService,
+            interceptedStudyViewFilter,
+            List.of(xAxisAttributeId, yAxisAttributeId),
+            false
```

Comment:
Not sure why we filter empty clinical data for violin plot but not for density plot. Maybe we should filter empty data here as well and get rid of this additional parameter.